### ISSUE TITLE
Calling sharing_add_footer after (!) wp_print_footer_scripts in wp_footer to ensure jQuery dependancy

### ISSUE THREAD

**Issue by kasparsd** [Automattic/jetpack#1223]
Currently `sharing_add_footer` is being attached to `wp_footer` with priority 10 which is **before** WordPress has printed the footer scripts as `wp_print_footer_scripts` is attached to `wp_footer` with priority 20.

`sharing_add_footer` echoes inline Javascript during [`display_footer()`](https://github.com/Automattic/jetpack/blob/8ec83606e531c70d963291b591c8aff682775e6e/modules/sharedaddy/sharing-service.php#L548) ([here is an example](https://github.com/Automattic/jetpack/blob/6c49efab93f08f68780575f43936b546db477d54/modules/sharedaddy/sharing-sources.php#L499)) which depends on jQuery being already available. This fails in cases when jQuery is enqueued in the footer.

![jquery-error](https://cloud.githubusercontent.com/assets/169055/4791088/4e1a7528-5dd9-11e4-961f-6b0c929646d3.png)

The solution is to move [`sharing_add_footer`](https://github.com/Automattic/jetpack/blob/4ae9a0e4fe15877340da66edabb3235548e328c2/modules/sharedaddy/sharing-service.php#L618) to a later priority:

```
add_action( 'wp_footer', 'sharing_add_footer', 25 );
```

**Comment by CFXd** [comment]
@kasparsd @ronalfy 

Just wanted to bump this, since it's causing jQuery console errors in themes with jQuery properly enqueued in the footer.

Perhaps we can get a temporary fix as @kasparsd suggested of changing the action priority?

My git is acting funny but a PR would be much appreciated!

**Comment by dboulet** [comment]
I’m still seeing this issue with Jetpack 3.9.6 and WP 4.5. Changing the action priority of `sharing_add_footer()` won’t work, because it itself enqueues another script.

In WP 4.5, there is `wp_add_inline_script()`, which could possibly be used to fix this. Is that a doable? Is Jetpack required to support older versions of WP?

https://developer.wordpress.org/reference/functions/wp_add_inline_script/

**Comment by jeherve** [comment]
> Is Jetpack required to support older versions of WP?

Yes, we aim for Jetpack to be compatible with the current version and one major version before that, so we'll need to wait for WordPress 4.6 to be released before to require 4.5+.

**Comment by jeherve** [comment]
Also reported in 59964-zd-vip

**Comment by kasparsd** [comment]
I updated my original report with a link to the [actual script from `Share_Email`](https://github.com/Automattic/jetpack/blob/6c49efab93f08f68780575f43936b546db477d54/modules/sharedaddy/sharing-sources.php#L499) which calls the jQuery directly.

**Comment by grappler** [comment]
> Yes, we aim for Jetpack to be compatible with the current version and one major version before that, so we'll need to wait for WordPress 4.6 to be released before to require 4.5+.

@jeherve 4.7 has been released so we could create a PR for this?

**Comment by kasparsd** [comment]
@jeherve I think the JS `document.getElementById('jetpack-source_f_name').value = ''` isn't needed anymore since the value of that field is no longer set to `hello`.

**Comment by jeherve** [comment]
@grappler Yes. It looks like @kasparsd already took care of it (thank you!). You can test #6339 and give your feedback there if you'd like!

**Comment by jeherve** [comment]
Reopening as we had to revert this in #6649.

This was explained in #6641.

Next steps to get this fixed once and for all are described here:
https://github.com/Automattic/jetpack/pull/6641#issuecomment-286382055

**Comment by aheckler** [comment]
3128393-t

**Comment by lamdayap** [comment]
related: [link](https://wordpress.org/support/topic/jetpack-sharing-buttons-showing-properly-but-not-working/)
